src/client.cpp

```diff
@@ -1257,15 +1257,15 @@ void CClient::ProcessAudioDataIntern ( CVector<int16_t>& vecsStereoSndCrd )
             if ( bMuteOutStream )
             {
                 iUnused = opus_custom_encode ( CurOpusEncoder,
-                                               &vecZeros[i * iNumAudioChannels * iOPUSFrameSizeSamples],
+                                               &vecZeros[static_cast<size_t> ( i * iNumAudioChannels * iOPUSFrameSizeSamples )],
```

Comment:
I don't think this fixes it. The multiplication is still done at the smaller size, before casting the result to the larger size, which is already implicit in the `[]` operator.

To do it by casting, I think you only cast `i`, which makes the multiplication be done all at `size_t`:
```
...[(static_cast<size_t>(i)) * iNumAudioChannels * iOPUSFrameSizeSamples]
...
```

Reply:
Multiplication will be done in the type of the first operant. Nevertheless if the result is assigned to a smaller type there should still be a check to handle any overflows.